src/extension.ts

```diff
@@ -30,123 +36,187 @@ class Extension {
+    type: DeviceType
+  ) {
     const devicesToHide = newExcludedDevices.filter(
-      (current) => !this._lastExludedDevices!.includes(current)
+      (current) =>
+        !(
+          type === "output"
+            ? this._lastExcludedOutputDevices!
+            : this._lastExcludedInputDevices!
+        ).includes(current)
     );
 
-    const devicesToHideIds = this._mixer
-      ?.getAudioDevicesFromDisplayNames(devicesToHide)
+    return this._mixer
+      ?.getAudioDevicesFromDisplayNames(devicesToHide, type)
       .filter((n) => n)
       .map((n) => n!.id);
-    return devicesToHideIds;
   }
 
-  private getDeviceIdsToShow(newExcludedDevices: DisplayName[]) {
-    const devicesToShow = this._lastExludedDevices!.filter(
-      (last) => !newExcludedDevices.includes(last)
-    );
-    const devicesToShowIds = this._mixer
-      ?.getAudioDevicesFromDisplayNames(devicesToShow)
+  private getDeviceIdsToShow(
+    newExcludedDevices: DisplayName[],
+    type: DeviceType
+  ) {
+    const devicesToShow = (
+      type === "output"
+        ? this._lastExcludedOutputDevices!
+        : this._lastExcludedInputDevices!
+    ).filter((last) => !newExcludedDevices.includes(last));
+
+    return this._mixer
+      ?.getAudioDevicesFromDisplayNames(devicesToShow, type)
       .filter((n) => n)
       .map((n) => n!.id);
-
-    return devicesToShowIds;
   }
 
-  setupOutputChangesSubscription() {
+  setupDeviceChangesSubscription() {
     this._mixerSubscription =
-      this._mixer?.subscribeToOutputChanges((event) => {
-        this.updateAvailableOutputsInSettings(event);
+      this._mixer?.subscribeToDeviceChanges((event) => {
+        this.updateAvailableDevicesInSettings(event);
 
         if (event.type === "output-added") {
-          this.hideDeviceIfExcluded(event.deviceId);
+          this.hideDeviceIfExcluded(event.deviceId, "output");
+        } else if (event.type === "input-added") {
+          this.hideDeviceIfExcluded(event.deviceId, "input");
         }
       }) ?? null;
   }
 
-  hideDeviceIfExcluded(deviceId: number) {
+  hideDeviceIfExcluded(deviceId: number, type: DeviceType) {
     if (!this._mixer) {
       return;
     }
 
-    const deviceName = this._mixer.getAudioDevicesFromIds([deviceId])[0]
+    const deviceName = this._mixer.getAudioDevicesFromIds([deviceId], type)[0]
       .displayName;
 
-    const excludedOutputs = this._settings?.getExcludedOutputDeviceNames();
+    const excludedDevices =
+      type === "output"
+        ? this._settings?.getExcludedOutputDeviceNames()
+        : this._settings?.getExcludedInputDeviceNames();
 
-    if (excludedOutputs?.includes(deviceName)) {
+    if (excludedDevices?.includes(deviceName)) {
       delay(200).then(() => {
         // delay due to potential race condition with Quick Setting panel's code
-        this._audioPanel!.removeDevice(deviceId);
+        this._audioPanel!.removeDevice(deviceId, type);
       });
     }
   }
 
   setAllOutputsInSettings() {
```

Comment:
This function name needs to be updated

Reply:
Sorry, I don't understand what exactly needs to be updated here

Reply:
The function name is still ‘’setAllOutputsInSettings”, although now it handles both outputs and inputs.

Reply:
Oh. I see it now. I thought the issue was on the line that was changed. 

Unfortunately I won't be able to change it for the next few hours as I'm not at home right now. 

Feel free to rename it on your side if you want to merge it today.

Reply:
Alright :slightly_smiling_face: